server/user.go

```diff
@@ -131,6 +133,12 @@ func (s *UserMgmtServer) getUser(w http.ResponseWriter, r *http.Request, ps http
 		return
 	}
 
+	if s.allowClientCredsAuth && creds.User.ID != "" {
```

Comment:
Why is the block (and the identical other two) needed? This should absolutely be checked already by `getCreds`